### PR TITLE
[SegmentV2] Optimize the upgrade logic of SegmentV2

### PR DESCRIPTION
**Pull request by morningman**
This CL mainly made the following modifications:

1. Reorganized SegmentV2 upgrade document.
2. When the variable `use_v2_rollup` is set to true, the base rollup in v2 format is forcibly queried for verifying the data.
3. Fix a problem that there is no persistent storage format information in the schema change operation that performs v2 conversion.
4. Allow users to directly create v2 format tables.

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]
+1. LGTM. The doc logic is more clear!

File: docs/documentation/cn/administrator-guide/segment-v2-usage.md [hunk 1]
```diff
@@ -1,118 +1,128 @@
...
+# Segment V2 升级手册
 
 ## 背景
 
-Doris 0.12版本中实现了segment v2（新的存储格式），引入词典压缩、bitmap索引、page cache等优化，能够提升系统性能。目前0.12版本已经发布alpha版本，正在内部上线过程中，上线的方案和试用方法记录如下
+Doris 0.12 版本中实现了新的存储格式：Segment v2，引入词典压缩、bitmap索引、page cache等优化，能够提升系统性能。
+
+0.12 版本会同时支持读写原有的 Segment V1（以下简称V1） 和新的 Segment V2（以下简称V2） 两种格式。如果原有数据想使用 V2 相关特性，需通过命令将 V1 转换成 V2 格式。
+
+本文档主要介绍从 0.11 版本升级至 0.12 版本后，如果转换和使用 V2 格式。
```
Comment (morningman) [inline docs/documentation/cn/administrator-guide/segment-v2-usage.md:9]:
```suggestion
本文档主要介绍从 0.11 版本升级至 0.12 版本后，如何转换和使用 V2 格式。
```

File: fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java [hunk 2]
```diff
@@ -109,9 +109,22 @@ public BestIndexInfo selectBestMV(ScanNode scanNode) throws UserException {
         long start = System.currentTimeMillis();
         Preconditions.checkState(scanNode instanceof OlapScanNode);
         OlapScanNode olapScanNode = (OlapScanNode) scanNode;
+
+        ConnectContext connectContext = ConnectContext.get();
+        if (connectContext != null && connectContext.getSessionVariable().isUseV2Rollup()) {
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java:114]: If user set 'use v2 rollup', the best rollup will not be selected?

File: fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java [hunk 3]
```diff
@@ -109,9 +109,22 @@ public BestIndexInfo selectBestMV(ScanNode scanNode) throws UserException {
...
+            OlapTable tbl = olapScanNode.getOlapTable();
+            String v2RollupIndexName = MaterializedViewHandler.NEW_STORAGE_FORMAT_INDEX_NAME_PREFIX + tbl.getName();
+            Long v2RollupIndexId = tbl.getIndexIdByName(v2RollupIndexName);
+            if (v2RollupIndexId != null) {
+                return new BestIndexInfo(v2RollupIndexId, false, "use_v2_rollup is true");
+            }
+        }
+
         Map<Long, List<Column>> candidateIndexIdToSchema = predicates(olapScanNode);
         long bestIndexId = priorities(olapScanNode, candidateIndexIdToSchema);
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java:126]: Maybe We should check the candidate index firstly and choose the best rollup in priorities.

**Review by EmmyMiao87 — CHANGES_REQUESTED** [review]

File: docs/documentation/cn/administrator-guide/segment-v2-usage.md [hunk 4]
```diff
@@ -1,118 +1,128 @@
-# Doris Segment V2上线和试用手册
```
[the remark this answers is not in the document]
Comment (morningman) [reply docs/documentation/cn/administrator-guide/segment-v2-usage.md:1]: ok

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 5]
```diff
@@ -944,7 +945,9 @@ private void createJob(long dbId, OlapTable olapTable, Map<Long, LinkedList<Colu
             } else if (hasIndexChange) {
                 needAlter = true;
             } else if (storageFormat == TStorageFormat.V2) {
-                needAlter = true;
+                if (olapTable.getTableProperty().getStorageFormat() != TStorageFormat.V2) {
```
[the remark this answers is not in the document]
Comment (morningman) [reply fe/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:948]: Yes, if storage format is already V2, `needAlter` will remain `false`, and nothing will be done.

File: fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java (hunk 2 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java:114]: Yes, if 'use v2 rollup' is true, only base index will be selected

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**